### PR TITLE
Update to version 0.2

### PR DESCRIPTION
**Pull request by xgouchet**
This PR has been created automatically by the CI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +54,35 @@ import { DdRumReactNavigationTracking } from 'dd-sdk-reactnative';
...
+    id: "1337", 
+    name: "John Smith", 
+    email: "john@example.com", 
+    type: "premium"
+})
+```
+
+#### Global attributes
+
+You can also keep global attributes to track information about a specific session: A/B testing configuration, advert campaign origin, cart status, …
```
Comment (jtappa) [inline README.md:76]:
```suggestion
You can also keep global attributes to track information about a specific session, such as A/B testing configuration, advert campaign origin, or cart status.
```

File: README.md [hunk 2]
```diff
@@ -50,6 +54,35 @@ import { DdRumReactNavigationTracking } from 'dd-sdk-reactnative';
     </NavigationContainer>
 ```
 
+### Adding custom attributes
+
+In order to get more information out of your RUM sessions, you can attach user informations to all RUM events.
```
Comment (jtappa) [inline README.md:59]:
```suggestion
You can attach user information to all RUM events to get more detailed information from your RUM sessions. 
```

File: README.md [hunk 3]
```diff
@@ -50,6 +54,35 @@ import { DdRumReactNavigationTracking } from 'dd-sdk-reactnative';
...
+
+#### User information
+
+For user specifc information, you can use the following code wherever you want in your code (after the SDK has been initialized). Note that the `id`, `name` and `email` attributes are built-in the Datadog UI, but you can add any attribute that makes sense to your app.
```
Comment (jtappa) [inline README.md:63]:
```suggestion
For user specifc information, you can use the following code wherever you want in your code (after the SDK has been initialized). The `id`, `name` and `email` attributes are built into the Datadog UI, but you can add any attribute that makes sense to your app.
```

**Review by jtappa — CHANGES_REQUESTED** [review]

File: example/android/app/build.gradle [hunk 4]
```diff
@@ -116,7 +116,7 @@ def jscFlavor = 'org.webkit:android-jsc:+'
  * on project.ext.react, JavaScript will not be compiled to Hermes Bytecode
  * and the benefits of using Hermes will therefore be sharply reduced.
  */
-def enableHermes = project.ext.react.get("enableHermes", false);
+def enableHermes = false;//project.ext.react.get("enableHermes", false);
```
Comment (mariusc83) [inline example/android/app/build.gradle:119]: you want to keep this like that ?

File: example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java [hunk 5]
```diff
@@ -34,7 +34,7 @@ public boolean getUseDeveloperSupport() {
           List<ReactPackage> packages = new PackageList(this).getPackages();
           // Packages that cannot be autolinked yet can be added manually here, for DdSdkReactNativeExample:
           // packages.add(new MyReactNativePackage());
-          packages.add(new DdSdkReactNativePackage());
+          // packages.add(new DdSdkReactNativePackage());
```
Comment (mariusc83) [inline example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java:37]: Where are those packages coming from ?

File: ios/DdLogs.m [hunk 6]
```diff
@@ -4,7 +4,7 @@
  * Copyright 2016-Present Datadog, Inc.
  */
 
-@import React;
+#import <React/RCTBridgeModule.h>
```
Comment (buranmert) [inline ios/DdLogs.m:7]: does `@import React;` give errors? 
this change was a fix attempt when `RCTBridgeModule` caused compilation errors in Alex's local
i had the same problem once in my local but i couldn't reproduce it consistently

File: ios/DdLogs.swift [hunk 7]
```diff
@@ -9,11 +9,7 @@ import DatadogSDKBridge
 
 @objc(DdLogs)
 class RNDdLogs: NSObject {
-    @objc(requiresMainQueueSetup)
-    static func requiresMainQueueSetup() -> Bool {
-        return false
-    }
```
Comment (buranmert) [inline ios/DdLogs.swift:15]: absence of this method was causing warnings in debug console.
i'm guessing it still gives warnings as RN version isn't changed.
did you run it in iOS simulator?

File: example/index.tsx [hunk 8]
```diff
@@ -14,15 +16,22 @@ const config = new DdSdkReactNativeConfiguration(
...
 
 DdSdkReactNative.initialize(config).then(() => {
 
     DdLogs.info('The RN Sdk was properly initialized', {
         foo: 42,
         bar: 'xyz',
     })
+
+    DdSdk.setUser({id: "1337", name: "Xavier", email: "xg@example.com", type: "premium"})
+
```
Comment (mariusc83) [inline example/index.tsx:33]: it is quite confusing to call these methods from the `DdSdk` but the initialization from the `DdSdkReactnative`

**Review by mariusc83 — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -28,6 +28,10 @@ const config = new DdSdkReactNativeConfiguration(
     true, // track XHR Resources
     true // track Errors
 )
+// Optional: enable or disable native craash reports
```
Comment (ncreated) [inline README.md:31]: typo
```suggestion
// Optional: enable or disable native crash reports
```

File: example/index.tsx (hunk 8 above)
Comment (xgouchet) [reply example/index.tsx:33]: For me it seems overkill to artificially create a method in DdSdkReactnative that would add no value other than transferring the call to DdSdk. 
Besides for people manually instrumenting their code, they would not use the DdSdkReactnative class.

File: ios/DdLogs.m (hunk 6 above)
Comment (xgouchet) [reply ios/DdLogs.m:7]: This class is automatically generated from `dd-mobile-bridge` tasks. If this **needs** to be changed, we should change it there and run the generation code again. 

That being said: 
- I didn't get any issue when running my tests in iOS. 
- This is the official recommended import when building native modules

File: example/android/app/build.gradle (hunk 4 above)
Comment (xgouchet) [reply example/android/app/build.gradle:119]: No that was a test

File: example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java (hunk 5 above)
Comment (xgouchet) [reply example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java:37]: Auto generated when you create a Native Module.

File: ios/DdLogs.swift (hunk 7 above)
Comment (xgouchet) [reply ios/DdLogs.swift:15]: Again, this class is auto generated, if this change is important, we should make the change in the generator classes. 
And yes, I ran it in the iOS simulator, and it all worked.

Comment (buranmert) [reply ios/DdLogs.swift:15]: it works, it doesn't give an error; it gives a warning

**Review by mariusc83 — APPROVED** [review]

File: example/android/app/build.gradle [hunk 10]
```diff
@@ -210,13 +210,7 @@ dependencies {
         exclude group:'com.facebook.flipper'
     }
 
-    if (enableHermes) {
```
Comment (0xnm) [inline example/android/app/build.gradle:213]: there is a flag `enableHermes`, but it is not used after this change? maybe then it is worth to remove project property as well?

File: example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java [hunk 11]
```diff
@@ -34,7 +34,6 @@ public boolean getUseDeveloperSupport() {
           List<ReactPackage> packages = new PackageList(this).getPackages();
           // Packages that cannot be autolinked yet can be added manually here, for DdSdkReactNativeExample:
           // packages.add(new MyReactNativePackage());
-          packages.add(new DdSdkReactNativePackage());
```
Comment (0xnm) [inline example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java:37]: so now it is autolinked, I guess?

**Review by 0xnm — APPROVED** [review]

**Review by 0xnm — APPROVED** [review]

File: example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java (hunk 11 above)
Comment (xgouchet) [reply example/android/app/src/main/java/com/example/ddsdkreactnative/MainApplication.java:37]: Yes exactly

File: ios/DdLogs.swift (hunk 7 above)
Comment (buranmert) [reply ios/DdLogs.swift:15]: does that change require a new PR in `mobile-bridge` repo?

**Review by ncreated — APPROVED** [review]

**Review by jtappa — APPROVED** [review]

**Pull request merged**